test/test_va_api_fixture.cpp

```diff
@@ -367,6 +367,135 @@ void VAAPIFixture::doDestroyConfig()
     ASSERT_STATUS(vaDestroyConfig(m_vaDisplay, m_configID));
 }
 
+void VAAPIFixture::querySurfaceAttributes(SurfaceAttributes& attribs) const
+{
+    ASSERT_TRUE(attribs.empty())
+        << "test logic error: surface attributes must be empty";
+
+    unsigned numAttribs(0);
+
+    ASSERT_STATUS(vaQuerySurfaceAttributes(m_vaDisplay, m_configID, NULL,
+        &numAttribs));
+
+    ASSERT_GT(numAttribs, 0u);
+
+    attribs.resize(numAttribs);
+
+    ASSERT_STATUS(vaQuerySurfaceAttributes(m_vaDisplay, m_configID,
+        attribs.data(), &numAttribs));
+
+    ASSERT_GT(numAttribs, 0u);
+    EXPECT_GE(attribs.size(), numAttribs);
+
+    attribs.resize(numAttribs);
+
+    const uint32_t flags = 0x0 | VA_SURFACE_ATTRIB_GETTABLE
+        | VA_SURFACE_ATTRIB_SETTABLE;
+
+    for (const auto& attrib : attribs) {
+        EXPECT_NE(attrib.flags & flags,
+            (uint32_t)VA_SURFACE_ATTRIB_NOT_SUPPORTED);
+        EXPECT_GE(attrib.value.type, VAGenericValueTypeInteger);
+        EXPECT_LE(attrib.value.type, VAGenericValueTypeFunc);
```

Comment:
I could not find reference to these in libva spec.

Reply:
These checks just ensure that the generic type is one of the VAGenericValueType enum values.  It is the same check used in the legacy doQuerySurfaceAttributes.

Reply:
The enum has VAGenericValueTypeFunc. No SurfaceAttributes are of type Func. So checking less than or equal to TypeFunc seems wrong. Not sure why was it introduced in legacy test, though. 
libva-utils tests should adhere to libva spec only, imo.

Reply:
Please go back and study the VASurfaceAttrib data structure again... it has a VAGenericValue member field called `value`.  The checks here are checking the `attrib.value.type`, not the `attrib.type`.

Reply:
My mistake. Peace.